### DISCUSSION TITLE
How to debug with breakpoints on Detox

### DISCUSSION THREAD

**Discussion by ggilchrist-ledger** [wix/Detox#3257] [Q&A]
Hi,

I'm fairly new to detox and on the homepage I noticed that Detox is setup to use breakpoints for debugging. 
![image](https://user-images.githubusercontent.com/87659842/157419598-8c0a71d9-fb69-4bd0-b6ef-67dd77e71512.png)

Playwright (for desktop E2E tests) has an inspector tool that allows you to step through the code with `await page.pause()` which makes it really easy to debug. Does Detox have a similar thing? Really I'm wondering how those familiar with Detox step through the code and if there's any tricks that could help. I know that VSCode has the ability to set breakpoints and step through JS code, perhaps it's just that that's being referred to? Thanks

**Comment by ggilchrist-ledger** [comment]
I found a nice simple guide here for anyone wondering. Yet to try it though: https://github.com/invertase/react-native-firebase/tree/main/tests#debugging-e2e-js-tests-vscode

**Reply by d4vidi** [reply]
Personally, I normally set up a jest runner on Webstorm that mimics Detox' behavior. It's pretty simple to do. You run detox once with `--loglevel verbose`, grab everything from the command to run `jest` in the command line, and compile it into the runner. Viola! you run the runner in debug mode, step-by-step -- the works.
